src/libAtomVM/opcodesswitch.h

```diff
@@ -6773,15 +6771,11 @@ HOT_FUNC int scheduler_entry_point(GlobalContext *glb)
                                 break;
                             }
                             case STRING_ATOM: {
-                                if (offset % 8) {
-                                    TRACE("bs_create_bin/6: current offset (%li) is not evenly divisible by 8\n", offset);
-                                    RAISE_ERROR(UNSUPPORTED_ATOM);
-                                }
-                                uint8_t *dst = (uint8_t *) term_binary_data(t) + (offset / 8);
+                                uint8_t *dst = (uint8_t *) term_binary_data(t);
                                 size_t remaining = 0;
                                 const uint8_t *str = module_get_str(mod, src_value, &remaining);
                                 segment_size = size_value * segment_unit;
-                                memcpy(dst, str, segment_size / 8);
+                                bitstring_copy_bits(dst, offset, str, segment_size);
```

Comment:
I think it would be better for performance reasons doing:
```c
    if (bits_offset % 8 == 0 && bits_count % 8 == 0) {
        memcpy(dst + (bits_offset / 8), src, bits_count / 8);
    } else {
        bitstring_copy_bits(dst, offset, str, segment_size);
    }
```

So we reduce function calls for that best case scenario.

Reply:
I chose to inline `bitstring_copy_bits` and have the partial case in a separate and non-inline function (`bitstring_copy_bits_incomplete_bytes`).